### ISSUE TITLE
doxygen x_noenv should always diff system-dependent settings

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9236]
**Describe the bug**
There are some settings in the Doxyfile that are system-dependent, like `CASE_SENSE_NAMES`. These are not properly identified when x_noenv is run.

**Expected behavior**
When doxygen -x_noenv is run, it should always print these values, whatever its value, as one is assuming no environment (in a broad sense, one does not want to be system-dependent). Or maybe print them under special section with asterisks, with a footnote that this might be a "system dependent diff".

The alternative would be that the default value of all variables are made always the same for all systems.

**To Reproduce**
```
doxygen -g
doxygen -x_noenv
CASE_SENSE_NAMES should appear there as 'diff'
```

**Version**
Doxyfile 1.9.4 (91095fb62f0bc2c624b0e8e49d9316f51be76559*)

**Additional context**
https://github.com/root-project/root/pull/9966

**Comment by albert-github** [comment]
You wrote:
> There are some settings in the Doxyfile that are system-dependent

Besides the `CASE_SENSE_NAMES` which other ones did you identify? I could only find the `CASE_SENSE_NAMES`.
Other system dependent values like `NUM_PROC_THREADS` are / should be "protected" by a 0 value (as the minval).

For the boolean value of  `CASE_SENSE_NAMES` this is not possible as we have here only 2 possible values (false and true) and nothing to use as a flag. Furthermore the determination of the value to be used is set, by means of a function call, during the initialization of the config structure (though at that moment the value of the option is  known). The mean problem remains that the bool value only has 2 possibilities, maybe we have to switch here as well to an enum value with the values `NO`, `YES` and `SYSTEM`, where `SYSTEM` determines the value at **runtime**.
(Note the current `defval` has no real meaning here as always the function is called).

**Comment by ferdymercury** [comment]
> Besides the `CASE_SENSE_NAMES` which other ones did you identify? I could only find the `CASE_SENSE_NAMES`.

True, there is only this one.



> For the boolean value of `CASE_SENSE_NAMES` this is not possible as we have here only 2 possible values

I think I did not explain it clearly. What I meant is that, when you run doxygen -x_noenv, it always shows the CASE_SENSE_NAMES boolean stored in the Doxyfile. Independently on its value. To ensure that if I make a diff on Macos in a PR and you try it on Linux, we are using the same value, YES or NO.

**Comment by ferdymercury** [comment]
To give an example, in our ROOT Doxyfile, we had CASE_SENSE_NAMES set to yes. When I run doxygen -x_noenv in Linux, it didn't appear in the diff, as it was thought to be the default value. Thus I wrote another Doxyfile only specifying the non-default values. Now I moved to macOS, and because its default value for this variable is different, doxygen was running with different settings. Thus, I think that:
- if CASE_SENSE_NAMES is defined in the Doxyfile, it should always print its value in the -x_noenv version
- if not defined, then it should not

I'd say we do not need a third SYSTEM option, as this would be the same as "leaving it out" of the Doxyfile ?

**Comment by albert-github** [comment]
The MacOS and Linux have different defaults (as far as I can see)
Form `portable.cpp`:
```
bool Portable::fileSystemIsCaseSensitive()
{
#if defined(_WIN32) || defined(macintosh) || defined(__MACOSX__) || defined(__APPLE__) || defined(__CYGWIN__)
  return FALSE;
#else
  return TRUE;
#endif
}
```
so for MacOS the default is `FALSE` (`NO`) and for Linux the default is `TRUE` (`YES`).

It might be that these definitions are outdated and the MacOS is case sensitive as well (like Linux).
In case MacOS is case sensitive now we might need to improve the default setting here.

Regarding:
>if CASE_SENSE_NAMES is defined in the Doxyfile, it should always print its value in the -x_noenv version
>if not defined, then it should not

this will be difficult as at the moment of printing it is not known anymore whether or not the value was in the used doxygen settings file or not.

**Comment by ferdymercury** [comment]
> In case MacOS is case sensitive now we might need to improve the default setting here.

Sounds good.



> this will be difficult as at the moment of printing it is not known anymore whether or not the value was in the used doxygen settings file or not.

oh, I see.

**Comment by albert-github** [comment]
Is MacOS currently case sensitive (I cannot test this as I don't have access to any MacOS system).

**Comment by ferdymercury** [comment]
I don't have access to it either. But I checked online and it seems it's by default not case sensitive, but it can be configured to be. So maybe the default doxygen setting should indeed not be changed.

**Comment by albert-github** [comment]
@doxygen what is your opinion?

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9526

**Comment by ferdymercury** [comment]
Thanks!!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.5.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
